### DISCUSSION TITLE
Azure Monitor: extract metric dimensions as variables

### DISCUSSION THREAD

**Discussion by secustor** [grafana/grafana#38332] [Data Sources]
**What would you like to be added**:
I would like to see an additional function in the style of `MetricNames(aResourceGroup, aNamespace, aResourceName)` which returns the available dimensions values for a specific metric. 

Suggested functions: 
- DimensionNames(?aSubscription?, aResourceGroup, aNamespace, aResourceName, aMetricName)
- DimensionValues(?aSubscription?, aResourceGroup, aNamespace, aResourceName, aMetricName, aDimensionKey)

**Why is this needed**:

My use case is the creation of repeating rows based on the available dimensions values

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by andresmgot** [comment]
Related to #39970. Once ARG is available as a query type there people could create their own variables.

**Comment by taleena** [comment]
Can track ARG being added in: https://github.com/grafana/grafana/issues/50233. :)
